sys/net/sock/sock_util.c

```diff
@@ -147,6 +150,45 @@ int sock_urlsplit(const char *url, char *hostport, char *urlpath)
     return 0;
 }
 
+int _parse_port(sock_udp_ep_t *ep_out, const char *portstart)
+{
+    int port_len = strlen(portstart);
+
+    /* Checks here verify that the supplied port number is up to 5 (random)
+     * chars in size and result is smaller or equal to UINT16_MAX. */
+    if (port_len > PORT_STR_LEN) {
+        return -EINVAL;
+    }
+    uint32_t port = atol(portstart);
+    if (port > UINT16_MAX) {
+        return -EINVAL;
+    }
+    ep_out->port = (uint16_t)port;
+    return port_len;
+}
+
+int _parse_netif(sock_udp_ep_t *ep_out, char *netifstart)
+{
+    char *netifend;
+    size_t netiflen;
+    char netifbuf[NETIF_STR_LEN + 1];
+
+    for (netifend = netifstart; *netifend && *netifend != ']';
+         netifend++);
+    netiflen = netifend - netifstart;
+    if (!*netifend || (netiflen >= NETIF_STR_LEN) || (netiflen == 0)) {
+        /* no netif found, bail out */
+        return -EINVAL;
+    }
+    strncpy(netifbuf, netifstart, netiflen);
+    int netif = strtol(netifbuf, NULL, 10);
+    if ((netif > UINT16_MAX) || (netif < 0)) {
```

Comment:
```suggestion
    if ((netif > INT16_MAX) || (netif < 0)) {
```

Reply:
but the `netif` member of a `sock_udp_ep_t` is `uint16_t`...

Reply:
> ```
> [01m[Ksock_util.c:185:16:[m[K [01;31m[Kerror: [m[Kcomparison between signed and unsigned integer expressions [-Werror=sign-compare]
>      if ((netif > UINT16_MAX) || (netif < 0)) {
> ```

Reply:
So I should cast to the smaller scope? I think it makes more sense to cast `netif` to `(unsigned)` ;-).